components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -148,6 +148,21 @@ struct ModelWithSubtitle {
   string subtitle;
 };
 
+struct ActionDetails {
+  ActionType type;
+};
+
+struct LabeledAction {
```

Comment:
Sorry to be a pain, but wouldn't this have been the ideal place for a union?

```
union MenuEntry {
    LabeledAction action;
    string subheading;
}
```

and then you wouldn't need to worry about `ActionDetails` being optional (as it would always be specified) when the object was a LabeledAction?

Reply:
i don't think we would even need `ActionDetails` anymore?

```c++
struct LabeledAction {
 string label;
 ActionType type;
}

union MenuEntry {
 LabeledAction action;
 string subheading;
}
```

in JS, we would still require a runtime check to determine whether you're dealing with an `action` or `subheading`. currently, i'm determining this by checking `actionDetails` exists or not. however, i do think your solution is more readable. i have also noticed that creating mojo structs in c++, especially unions, isn't clean at all as oppose to an optional field.

```js
context.actionList.map((actionGroup) => {
  <div>{actionGroup.category}</div>
  
  {actionGroup.entries.map((entry) => {
    if (entry.subheading) { // render subheading }
    else { // render LabeledAction } 
  })
})}
```

Reply:
ok, so i ended up with a union since it's cleaner. the following commit should express my seal:
<img width="288" alt="image" src="https://github.com/brave/brave-core/assets/8665427/3666d243-e77f-48a9-b124-3d212d76efc1">